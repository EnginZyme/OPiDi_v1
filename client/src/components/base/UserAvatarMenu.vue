<template>
  <div class="user_avatar_menu">
    <OverlayPanel
      ref="user_avatar_menu__overlay"
      appendTo="body"
      class="user_avatar_menu__overlay"
    >
      <SignOutView />
    </OverlayPanel>
    <div
      @click="toggleSignOutViewVisibility"
      aria:haspopup="true"
      id="user_avatar_menu__toggle_button"
    >
      <UserAvatarImage />
    </div>
  </div>
</template>

<script>
import UserAvatarImage from "./UserAvatarImage";
import SignOutView from "../views/SignOutView";

/**
 * A component to display the image of the authenticated user and to trigger the display of the
 * sign out view when clicked upon
 */
export default {
  name: "UserAvatarMenu",
  components: {
    UserAvatarImage,
    SignOutView,
  },
  methods: {
    /**
     * Triggers the opening/closing of the sign out view.
     * 
     * @param {object} event - User initiated toggle event.
     */
    toggleSignOutViewVisibility(event) {
      this.$refs.user_avatar_menu__overlay.toggle(event);
    },
  },
};
</script>

<style scoped>
.user_avatar_menu__overlay {
  background: #efefff;
  box-shadow: 2px 2px 9px rgba(0, 0, 0, 0.25);
  border-radius: 5px;
  width: 271px;
}
.user_avatar_menu {
  display: grid;
  align-content: center;
  cursor: pointer;
}
</style>
