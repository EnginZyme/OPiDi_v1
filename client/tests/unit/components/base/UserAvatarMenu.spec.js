import { shallowMount } from "@vue/test-utils";
import UserAvatarMenu from "@/components/base/UserAvatarMenu";

describe("shallowMounted UserAvatarMenu", () => {
  const wrapper = shallowMount(UserAvatarMenu);

  it("renders correctly", () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
