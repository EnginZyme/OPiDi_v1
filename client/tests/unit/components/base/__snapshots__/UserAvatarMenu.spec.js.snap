// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallowMounted UserAvatarMenu renders correctly 1`] = `
<div
  class="user_avatar_menu"
>
  <overlaypanel
    appendto="body"
    class="user_avatar_menu__overlay"
  >
    <signoutview-stub />
  </overlaypanel>
   
  <div
    aria:haspopup="true"
    id="user_avatar_menu__toggle_button"
  >
    <useravatarimage-stub
      size="small"
    />
  </div>
</div>
`;
